core/src/wsv.rs

```diff
@@ -447,36 +417,49 @@ impl<W: WorldTrait> WorldStateView<W> {
         &self,
         f: impl FnOnce(&World) -> Result<WorldEvent, Error>,
     ) -> Result<(), Error> {
-        let mut cur_events: SmallVec<[Event; 3]> = SmallVec::new();
-        let event = f(&self.world)?;
+        let world_event = f(&self.world)?;
 
-        match event {
+        match world_event {
             WorldEvent::Domain(domain_event) => {
                 match &domain_event {
                     DomainEvent::Account(account_event) => {
                         if let AccountEvent::Asset(asset_event) = account_event {
-                            cur_events.push(DataEvent::Asset(asset_event.clone()).into())
+                            let event = DataEvent::Asset(asset_event.clone());
```

Comment:
Macro? Generic helper?

Reply:
Another thing is that this is probably a good candidate for a `impl TryFrom<WorldEvent>`